Rocky.class.nut

```diff
@@ -565,16 +579,38 @@ class Rocky.Context {
     }
 
     function send(code, message = null, forcejson = false) {
-        // Cancel the timeout
-        if (timer) {
-            imp.cancelwakeup(timer);
-            timer = null;
-        }
+        _doSend(code, message, forcejson);
 
         // Remove the context from the store
         if (id in _contexts) {
             delete Rocky.Context._contexts[id];
         }
+    }
+
+    function setTimeout(timeout, callback) {
+        // Set the timeout timer
+        if (timer) imp.cancelwakeup(timer);
+        timer = imp.wakeup(timeout, function() {
+            if (callback == null) {
+                send(502, "Timeout");
```

Comment:
I think timeout is 408, isn't it? 502 is bad gateway.

Or is it about something else?

Reply:
This isn't something I changed. I just changed functions near it and Git showed that whole section as changed.